"use strict";(self.webpackChunklukemurraynz=self.webpackChunklukemurraynz||[]).push([[27848],{88388:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>u,contentTitle:()=>s,default:()=>d,frontMatter:()=>a,metadata:()=>r,toc:()=>c});var r=n(25921),o=n(74848),i=n(28453);const a={title:"Azure Function App Host Runtime Error",metaDescription:"Troubleshoot and resolve Azure Function App host runtime errors like BadGateway and InternalServerError due to storage account issues.",tags:["Azure"],categories:["Azure"],authors:["Luke"],slug:"azure/function-app-host-runtime",keywords:["Azure","Azure Functions","Function App","Host Runtime Error","BadGateway","InternalServerError","Storage Account","Networking","Private Endpoint","VNET Integration"],description:"Learn to resolve Azure Function App host runtime errors like BadGateway, often caused by storage account configuration issues.",date:new Date("2025-08-22T00:59:05.219Z")},s=void 0,u={authorsImageUrls:[void 0]},c=[];function l(e){const t={blockquote:"blockquote",code:"code",img:"img",li:"li",p:"p",ul:"ul",...(0,i.R)(),...e.components};return(0,o.jsxs)(o.Fragment,{children:[(0,o.jsx)(t.p,{children:"You might be deploying to Azure Functions or attempting to retrieve the Function App URL, and you may encounter errors like:"}),"\n",(0,o.jsxs)(t.blockquote,{children:["\n",(0,o.jsxs)(t.ul,{children:["\n",(0,o.jsx)(t.li,{children:(0,o.jsx)(t.code,{children:"Encountered an error (BadGateway) from host runtime."})}),"\n",(0,o.jsx)(t.li,{children:(0,o.jsx)(t.code,{children:"Encountered an error (InternalServerError) from host runtime."})}),"\n"]}),"\n"]}),"\n","\n",(0,o.jsx)(t.p,{children:(0,o.jsx)(t.img,{alt:"Host runtime",src:n(67348).A+"",width:"1730",height:"453"})}),"\n",(0,o.jsx)(t.p,{children:"I recently encountered an issue with a Function App deployment I was working on. It was working in an Azure CLI deployment, but as part of transitioning the Azure CLI to Bicep, I encountered an error when I noticed that my Timer trigger Function was not firing as expected. After some investigation, I found that the issue was due to the Storage account that the Function App was using. The Storage account was not correctly configured or accessible, which led to the host runtime errors."}),"\n",(0,o.jsxs)(t.blockquote,{children:["\n",(0,o.jsx)(t.p,{children:"The issue was with the Networking. Although I had deployed the storage account, I had failed to specify whether the storage account was to be publicly accessible or not, i.e., reachable by the Function App. So, Public access was off, and the Function App could not access the Storage account, which was causing the errors."}),"\n"]}),"\n",(0,o.jsx)(t.p,{children:"To remediate this, I enabled the public endpoint, validated it worked, then added the Storage account to a Virtual Network with Private endpoints, and enabled VNET (Virtual Network) integration to the Function App, allowing a secure connection between the function app and the storage account."})]})}function d(e={}){const{wrapper:t}={...(0,i.R)(),...e.components};return t?(0,o.jsx)(t,{...e,children:(0,o.jsx)(l,{...e})}):l(e)}},67348:(e,t,n)=>{n.d(t,{A:()=>r});const r=n.p+"assets/images/FunctionApp_HostRuntimeError-0de68ffd600f4eca1ecf93d2a68a9afc.jpg"},28453:(e,t,n)=>{n.d(t,{R:()=>a,x:()=>s});var r=n(96540);const o={},i=r.createContext(o);function a(e){const t=r.useContext(i);return r.useMemo((function(){return"function"==typeof e?e(t):{...t,...e}}),[t,e])}function s(e){let t;return t=e.disableParentContext?"function"==typeof e.components?e.components(o):e.components||o:a(e.components),r.createElement(i.Provider,{value:t},e.children)}},25921:e=>{e.exports=JSON.parse('{"permalink":"/azure/function-app-host-runtime","source":"@site/blog/2025-08-22-funcapphosterror/index.mdx","title":"Azure Function App Host Runtime Error","description":"Learn to resolve Azure Function App host runtime errors like BadGateway, often caused by storage account configuration issues.","date":"2025-08-22T00:59:05.219Z","tags":[{"inline":true,"label":"Azure","permalink":"/tags/azure"}],"readingTime":1.13,"hasTruncateMarker":true,"authors":[{"name":"Luke Murray","title":"Author","url":"https://luke.geek.nz","imageURL":"https://luke.geek.nz/img/logo.png","key":"Luke","page":null}],"frontMatter":{"title":"Azure Function App Host Runtime Error","metaDescription":"Troubleshoot and resolve Azure Function App host runtime errors like BadGateway and InternalServerError due to storage account issues.","tags":["Azure"],"categories":["Azure"],"authors":["Luke"],"slug":"azure/function-app-host-runtime","keywords":["Azure","Azure Functions","Function App","Host Runtime Error","BadGateway","InternalServerError","Storage Account","Networking","Private Endpoint","VNET Integration"],"description":"Learn to resolve Azure Function App host runtime errors like BadGateway, often caused by storage account configuration issues.","date":"2025-08-22T00:59:05.219Z"},"unlisted":false,"prevItem":{"title":"MCP Server with Azure Functions & Communication Services","permalink":"/azure/communication-services-mcp-email-server"},"nextItem":{"title":"Secure Isolated Recovery Environment in Azure","permalink":"/azure/sire/"}}')}}]);